chromium/popup.js

```diff
@@ -77,12 +77,34 @@ function createRuleLine(ruleset) {
   return line;
 }
 
+// Change the UX to reflect extension enabled/disabled
+function updateEnabledDisabledButtonState() {
+  document.getElementById('onoffswitch').checked = backgroundPage.isExtensionEnabled;
+}
+
+// Toggle extension enabled/disabled status
+function toggleEnabledDisabled() {
+  if (backgroundPage.isExtensionEnabled) {
+    // User wants to disable us
+    backgroundPage.isExtensionEnabled = false;
+    chrome.browserAction.setBadgeText({ text: "OFF" });
+  } else {
+    // User wants to enable us
+    backgroundPage.isExtensionEnabled = true;
+    chrome.browserAction.setBadgeText({ text: "" });
+  }
+   updateEnabledDisabledButtonState();
+  // The extension state changed, to reload this tab.
+  chrome.tabs.reload();
+}
+
 /**
  * Create the list of rules for a specific tab
- * @param tab
+ * @param tabArray
  */
-function gotTab(tab) {
-  var rulesets = backgroundPage.activeRulesets.getRulesets(tab.id);
+function gotTab(tabArray) {
+  var activeTab = tabArray[0];
```

Comment:
We are -- since this is called from the popup extension window.

If Chrome can't find the active rules popup window, something is catastrophically wrong -- and I don't think we can/should work around it.